lib/matic-jwt/grape/middleware/auth.rb

```diff
@@ -0,0 +1,53 @@
+module MaticJWT
+  module Grape
+    module Middleware
+      class Auth
+        def initialize(app, options)
+          @app = app
+          @secret_reader = options[:secret]
+        end
+
+        def call(env)
+          @env = env
+
+          validate_request
+          authenticate!
+          continue!
+        end
+
+        private
+
+        def validate_request
+          raise JWT::VerificationError, 'Authorization token is invalid' unless request.valid?
+        end
+
+        def authenticate!
+          @env['auth_payload'] = authenticate_with_jwt!(secret)
+        end
+
+        def continue!
+          @app.call(@env)
+        end
+
+        def authenticate_with_jwt!(secret)
+          payload = jwt_authenticator.authenticate_with_secret!(secret)
```

Comment:
Do I understand correctly that if request not authenticated - payload will be missing? I can come up with few usecases when we need payload regardless of authentication. However, they are not critical

Reply:
Yes, you understood correctly. Will fix

Reply:
Fixed